### PR TITLE
feat: add block endpoints

### PR DESCRIPTION
**Pull request by nathanielc**
Adds endpoints for direct storage and retrieval of blocks.

### CODE REVIEW

File: ceramic-kubo-rpc/src/block.rs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+/// Store a block into IFPS.
+#[tracing::instrument(skip_all)]
+pub async fn put<T, C>(client: T, codec: C, data: Vec<u8>) -> Result<Cid, Error>
+where
+    T: IpfsDep,
+    C: Codec,
+{
+    let hash = Code::Sha2_256.digest(&data);
+    let cid = Cid::new_v1(codec.into(), hash);
+    client.put(cid, data.into(), vec![]).await?;
```
Comment (smrz2001) [inline ceramic-kubo-rpc/src/block.rs:30]: Wouldn't this return the CID so you don't have to hash the data above?

File: ceramic-kubo-rpc/src/http/block.rs [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+    data: web::Data<AppState<T>>,
+    query: web::Query<PutQuery>,
+    mut payload: Multipart,
+) -> Result<HttpResponse, Error>
+where
+    T: IpfsDep,
+{
+    if query.pin {
+        return Err(Error::Invalid(anyhow!(
+            "recursive pinning is not supported."
```
Comment (smrz2001) [inline ceramic-kubo-rpc/src/http/block.rs:83]: Does `pin` being `true` here always mean recursively?

Comment (nathanielc) [reply ceramic-kubo-rpc/src/http/block.rs:83]: Yes, according to their docs https://docs.ipfs.tech/reference/kubo/rpc/#api-v0-block-put We might need to implement recursive pin but I am hoping not too. So much of our data points back to the beginning of time that its dangerous to recursively pin so for now I have explicitly denied any recursive pin behavior.

File: ceramic-kubo-rpc/src/block.rs (hunk 1 above)
Comment (nathanielc) [reply ceramic-kubo-rpc/src/block.rs:30]: The beetle API needs a CID in order to put the data in the first place.

**Review by zachferland — APPROVED** [review]

File: ceramic-kubo-rpc/src/block.rs (hunk 1 above)
Comment (smrz2001) [reply ceramic-kubo-rpc/src/block.rs:30]: Oh ok, huh.

**Review by smrz2001 — APPROVED** [review]
🚢

**Pull request merged**